src/python/pants/engine/target.py

```diff
@@ -438,6 +438,16 @@ def __eq__(self, other: Union[Target, Any]) -> bool:
             other.field_values,
         )
 
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, Target):
+            return NotImplemented
+        return self.address < other.address
```

Comment:
I note that I think pair of targets `a` and `b` can theoretically have all of these false: `a < b`, `a == b`, `a > b`, if the addresses are equal but at least one of the other values used in `__eq`__ (`__class__`, `residence_dir` or `field_values`) are not. That is, targets are not "totally ordered".

Does that matter?

Reply:
I considered that, but no I don't think that matters, as in our case, if the address is equal, it should be the same target, i.e. have the same field values too. So in this case, it's an optimization to not also compare the fields for the lt/gt methods used for sorting.